modules/apps/marketplace/marketplace-settings-web/test/js/MarketplaceConnect.spec.tsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {
+	cleanup,
+	fireEvent,
+	render,
+	waitForElementToBeRemoved,
+} from '@testing-library/react';
+import React from 'react';
+import {act} from 'react-dom/test-utils';
+
+import {MarketplaceSettings} from '../../src/main/resources/META-INF/resources/js';
+
+import '@testing-library/jest-dom/extend-expect';
+
+const marketplaceSettingsProps = {
+	baseResourceURL: 'http://localhost:8080?p_p_id=marketplace_settings',
+	clientId: 'marketplace-client-id',
+	portletNamespace: '_marketplacePortletNamespace_',
+	redirect: '/web/markeplace/authorize',
+	url: 'https://marketplace.liferay.com',
+};
+
+const messageEventInvalid = new MessageEvent('message', {
+	data: {
+		code: 'marketplace-uuid',
+	},
+	origin: 'https://marketplace.notLiferay.com',
+	source: window,
+});
+
+const messageEventInvalidWithoutData = new MessageEvent('message', {
+	origin: 'https://marketplace.liferay.com',
+	source: window,
+});
+
+const messageEventValid = new MessageEvent('message', {
+	data: {
+		code: 'marketplace-uuid',
+		serviceURL: 'https://springboot.marketplace.liferay.com',
+		settings: {},
+	},
+	origin: 'https://marketplace.liferay.com',
+	source: window,
+});
+
+const MarketplaceSettingsWrapper: React.FC<{
+	children?: React.ReactNode | undefined;
+}> = () => {
+	return (
+		<MarketplaceSettings
+			{...marketplaceSettingsProps}
+			learnResources={[]}
+		/>
+	);
+};
+
+describe('MarketplaceSettings', () => {
```

Comment:
My bad, this should be the same as the file name

```suggestion
describe('MarketplaceConnect', () => {
```